src/libaktualizr/storage/sqlstorage.cc

```diff
@@ -1648,6 +1648,48 @@ void SQLStorage::clearInstallationResults() {
   db.commitTransaction();
 }
 
+void SQLStorage::storeDeviceDataHash(const std::string& data_type, const std::string& hash) {
+  SQLite3Guard db = dbConnection();
+
+  auto statement = db.prepareStatement<std::string, std::string>(
+      "INSERT OR REPLACE INTO device_data(data_type,hash) VALUES (?,?);", data_type, hash);
+  if (statement.step() != SQLITE_DONE) {
+    LOG_ERROR << "Can't set " << data_type << " hash: " << db.errmsg();
+    return;
```

Comment:
Should we throw an exception here?

Reply:
Yes, probably.

Reply:
Although, to be fair, there are lots of other parts of the storage that don't throw given the same circumstances.

Reply:
Yes, I know. Still, will be fewer places to fix in the future ;)